lib/index.js

```diff
@@ -265,10 +288,16 @@ Metalsmith.prototype.write = unyield(function*(files, dir){
   dir = dir || this.destination();
   var write = this.writeFile.bind(this);
 
-  yield Object.keys(files).map(function(key){
-    var file = path.resolve(dir, key);
-    return write(file, files[key]);
-  });
+  var complete = 0;
+  var keys = Object.keys(files);
+  var concurrent = this.concurrent();
+  while (complete < keys.length) {
+    yield keys.slice(complete, complete + concurrent).map(function(key){
```

Comment:
Can we break this piece apart to be slightly easier to follow? Less one-liner-ish.